libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/roadname/RoadNameViewBinderTest.kt

```diff
@@ -0,0 +1,140 @@
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class, ExperimentalCoroutinesApi::class)
+@RunWith(RobolectricTestRunner::class)
+class RoadNameViewBinderTest {
+
+    @get:Rule
+    val loggerRule = LoggingFrontendTestRule()
+
+    @get:Rule
+    val coroutineRule = MainCoroutineRule()
+
+    private lateinit var ctx: Context
+    private lateinit var navContext: NavigationViewContext
+    private lateinit var mapboxNavigation: MapboxNavigation
+    private lateinit var loadedMapStyleFlow: MutableStateFlow<Style?>
+    private lateinit var sut: RoadNameViewBinder
+
+    @Before
+    fun setUp() {
+        ctx = ApplicationProvider.getApplicationContext()
+        navContext = spyk(
+            NavigationViewContext(
+                context = ctx,
+                lifecycleOwner = TestLifecycleOwner(),
+                viewModel = NavigationViewModel(),
+                storeProvider = { TestStore() }
+            )
+        )
+        loadedMapStyleFlow = MutableStateFlow(mockk())
+        every { navContext.mapStyleLoader } returns mockk {
+            every { loadedMapStyle } returns loadedMapStyleFlow
+        }
+
+        mapboxNavigation = mockk(relaxed = true)
+
+        sut = RoadNameViewBinder(navContext)
+    }
+
+    @Test
+    fun `bind should inflate road name layout`() {
+        val rootLayout = FrameLayout(ctx)
+
+        sut.bind(rootLayout)
+
+        assertNotNull(
+            "Expected MapboxRoadNameView",
+            rootLayout.findViewById<MapboxRoadNameView>(R.id.roadNameView)
+        )
+    }
+
+    @Test
+    fun `bind should return ReloadingComponent that attaches RoadNameComponent`() =
+        runBlockingTest {
+            val rootLayout = FrameLayout(ctx)
+            val reloadComponent = sut.bind(rootLayout) as ReloadingComponent<*>
+            reloadComponent.onAttached(mapboxNavigation)
+
+            assertTrue(reloadComponent.childComponent is RoadNameComponent)
+        }
+
+    @Test
+    @Suppress("MaxLineLength")
+    fun `bind should return ReloadingComponent that reloads child component on Map Style change`() =
+        coroutineRule.runBlockingTest {
+            loadedMapStyleFlow.value = null
+            val rootLayout = FrameLayout(ctx)
+            val reloadComponent = sut.bind(rootLayout) as ReloadingComponent<*>
+            reloadComponent.onAttached(mapboxNavigation)
+            assertNull(reloadComponent.childComponent)
+
+            loadedMapStyleFlow.value = mockk()
+
+            assertNotNull(reloadComponent.childComponent)
+        }
+
+    @Test
+    @Suppress("MaxLineLength")
+    fun `bind should return ReloadingComponent that reloads child component on roadNameBackground change`() =
```

Comment:
We can also check that background is applied to the view.

Reply:
Let's first move that responsibility to RoadNameComponent and then add a test for it.